go.mod

```diff
@@ -188,16 +196,18 @@ require (
 	gopkg.in/yaml.v2 v2.4.0 // indirect
 	gopkg.in/yaml.v3 v3.0.1 // indirect
 	nhooyr.io/websocket v1.8.6 // indirect
+	pgregory.net/rapid v0.5.3 // indirect
 )
 
 replace (
 	// use cosmos keyring
 	github.com/99designs/keyring => github.com/cosmos/keyring v1.1.7-0.20210622111912-ef00f8ac3d76
 	// ics23 patch for dragonberry
 	github.com/confio/ics23/go => github.com/cosmos/cosmos-sdk/ics23/go v0.8.0
+
+	github.com/cosmos/cosmos-sdk => github.com/cosmos/cosmos-sdk v0.46.0-beta2.0.20221109114442-bcdf81cbaf8d
```

Comment:
I think this might be a mishap right here? PR title says you're setting the repo up for sdk v0.47, but you're currently using an outdated version of v0.46.

This also explains some errors you are getting regarding the empty gas config for example.

Reply:
its this branch here: https://github.com/cosmos/cosmos-sdk/tree/release/v0.47.x

the semgrep hasn't been bumped yet.